whois-api/src/main/java/net/ripe/db/whois/api/httpserver/dos/UpdateDoSFilterHolder.java

```diff
@@ -0,0 +1,43 @@
+package net.ripe.db.whois.api.httpserver.dos;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+
+import static jakarta.ws.rs.HttpMethod.GET;
+
+@Component
+public class UpdateDoSFilterHolder extends AbstractDoSFilterHolder {
+
+    private final String dosUpdatesMaxSecs;
+
+    public UpdateDoSFilterHolder(@Value("${dos.filter.enabled:false}") final boolean dosFilterEnabled,
+                                 @Value("${ipranges.trusted}") final String trustedIpRanges,
+                                 @Value("${dos.filter.max.updates:10}") final String dosUpdatesMaxSecs) {
+        super(dosFilterEnabled, trustedIpRanges);
+
+        this.dosUpdatesMaxSecs = dosUpdatesMaxSecs;
+        this.setInitParameter("maxRequestMs", getMaxRequestPerms());
+        this.setInitParameter("maxRequestsPerSec", getMaxRequestPerSec());
+
+    }
+
+    @Override
+    protected boolean isAllowedMethod(final HttpServletRequest request) {
```

Comment:
Should be AND not OR ?

Reply:
Also  request != null &  (PUT.equalsIgnoreCase(request.getMethod()) || POST.equalsIgnoreCase(request.getMethod())) ?

Reply:
Should be isNotAllowed of change the logic.
I think that some refactor that I did make this unclear